ext/standard/var_unserializer.re

```diff
@@ -1201,6 +1201,111 @@ object ":" uiv ":" ["]	{
 	return object_common(UNSERIALIZE_PASSTHRU, elements, has_unserialize);
 }
 
+"E:" uiv ":" ["]	{
+	size_t len, maxlen;
+	char *str;
+	zend_string *enum_name, *case_name;
+	zend_class_entry *ce;
+
+	if (!var_hash) return 0;
+
+	len = parse_uiv(start + 2);
+	maxlen = max - YYCURSOR;
+	if (maxlen < len || len == 0) {
+		*p = start + 2;
+		return 0;
+	}
+
+	str = (char*)YYCURSOR;
+
+	YYCURSOR += len;
+
+	if (*(YYCURSOR) != '"') {
+		*p = YYCURSOR;
+		return 0;
+	}
+	if (*(YYCURSOR+1) != ';') {
+		*p = YYCURSOR+1;
+		return 0;
+	}
+
+	int32_t colon_pos = -1;
+	for (uint32_t i = 0; i < len; ++i) {
```

Comment:
```suggestion
	size_t colon_pos = SIZE_MAX;
	for (size_t i = 0; i < len; ++i) {
```

In recent php minor releases, strings can be longer than 4GB (this has been a common source of edge cases in the past) - on 64-bit platforms size_t is 8 bytes (and int32_t is obviously 4 bytes)

```
php > $x = str_repeat('a', 5_000_000_000);
php > echo strlen($x);
5000000000
```